tests/integration/test_manylinux.py

```diff
@@ -232,526 +186,608 @@ def assert_show_output(manylinux_ctr, wheel, expected_tag, strict):
-)
-@pytest.mark.parametrize('only_plat', [True, False])
-def test_build_wheel_compat(target_policy, only_plat, any_manylinux_container,
-                            docker_python, io_folder):
-    # test building wheels with compatibility with older spec
-    # check aliases for older spec
-
-    policy, tag, manylinux_ctr = any_manylinux_container
-
-    docker_exec(manylinux_ctr, [
-        'bash', '-c', 'cd /auditwheel_src/tests/integration/testsimple '
-                      '&& python -m pip wheel --no-deps -w /io .'
-    ])
-
-    filenames = os.listdir(io_folder)
-    orig_wheel = filenames[0]
-    assert 'manylinux' not in orig_wheel
-
-    if PLATFORM in {'x86_64', 'i686'}:
-        expect = f'manylinux_2_5_{PLATFORM}'
-        expect_tag = f'manylinux_2_5_{PLATFORM}.manylinux1_{PLATFORM}'
-    else:
-        expect = f'manylinux_2_17_{PLATFORM}'
-        expect_tag = f'manylinux_2_17_{PLATFORM}.manylinux2014_{PLATFORM}'
-
-    target_tag = target_policy
-    for pep600_policy, aliases in POLICY_ALIASES.items():
-        policy_ = f'{pep600_policy}_{PLATFORM}'
-        aliases_ = [f'{p}_{PLATFORM}' for p in aliases]
-        if target_policy == policy_ or target_policy in aliases_:
-            target_tag = f'{policy_}.{".".join(aliases_)}'
-
-    only_plat_arg = '--only-plat' if only_plat else ''
-    # we shall ba able to repair the wheel for all targets
-    docker_exec(manylinux_ctr, [
-        'bash', '-c',
-        f'auditwheel -v repair --plat {target_policy} {only_plat_arg} -w /io'
-        f' /io/{orig_wheel}'])
-    filenames = os.listdir(io_folder)
-    assert len(filenames) == 2
-    if only_plat or target_tag == expect_tag:
-        repaired_tag = target_tag
-    else:
-        repaired_tag = f'{expect_tag}.{target_tag}'
-    repaired_wheel = f'testsimple-0.0.1-{PYTHON_ABI}-{repaired_tag}.whl'
-    assert repaired_wheel in filenames
-    assert_show_output(manylinux_ctr, repaired_wheel, expect, True)
-
-    docker_exec(docker_python, f'pip install /io/{repaired_wheel}')
-    docker_exec(
-        docker_python,
-        ['python', '-c',
-         'from sys import exit; from testsimple import run; exit(run())']
-    )
+        # Repair the wheel using the appropriate manylinux container
+        repair_command = \
+            f'auditwheel repair --plat {policy} --only-plat -w /io /io/{orig_wheel}'
+        docker_exec(
+            manylinux_ctr,
+            ["bash", "-c", "LD_LIBRARY_PATH=/usr/local/lib " + repair_command]
+        )
+        filenames = os.listdir(io_folder)
+        assert len(filenames) == 2
+        repaired_wheel = f'nonpy_rpath-0.1.0-{PYTHON_ABI}-{tag}.whl'
+        assert repaired_wheel in filenames
+        assert_show_output(manylinux_ctr, repaired_wheel, policy, False)
+
+        # Test the resulting wheel outside the manylinux container
+        docker_exec(docker_python, "pip install /io/" + repaired_wheel)
+        if 'musllinux' in tag:
+            pytest.xfail("libcrypt segfaults")
```

Comment:
This should also be investigated. I didn't have time to do much for now.

Reply:
It segfaults because it's not using libxcrypt but musl embedded `crypt` function.

The `crypt` & `crypt_r` function are loaded from `libc.musl-*.so.1` because musl does not honor symbol version (the symbol that should be linked according to readelf is `crypt@@XCRYPT_2.0`). The musl `crypt` function just segfaults when passing a `NULL` salt.
libxcrypt should be removed from the musllinux image.